src/pages/Home.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+import Navbar from "../components/Navbar";
+import { apiBaseURL } from "../constants";
+import Spinner from "../components/Spinner";
+import VideoCard from "../components/VideoCard";
+
+export default function Home() {
+  const [loading, setLoading] = useState(true);
+  const [recommendedVideos, setRecommendedVideos] = useState([]);
+
+  useEffect(() => {
+    fetch(`${apiBaseURL}`)
+      .then((r) => r.json())
+      .then(setRecommendedVideos)
+      .finally(() => setLoading(false));
+  }, []);
```

Comment:
promise chaining is ugly and urs does no error handling; instead opt for await/async. Also why do you wrap `apiBaseURL` in a string template if ur not gonna do anything else w the template. It's just confusing syntax at that point

```javascript
useEffect(() => {
    async func fetchData() {
        const res = await fetch(apiBaseURL);
        if (res.status != 200) {
            alert("<put some error message and include the status code>");
        }
        
        const data = await res.json();
        setRecommendedVideos(data);

        setLoading(false);
    }

    fetchData();
}, []);
```

Reply:
or actually extract `fetchData` into a function outside useEffect then u can just have:

```javascript
useEffect(() => {
    fetchData();
}, []);
```

and you can explain when useEffect will be called in the comment above it

Reply:
Agreed, however your example also omits most error handling. Updated to isolate fetch functions to `api.js` and catch to components that use them.

Reply:
Feel free to reopen if you disagree with the change.